hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleDispatcher.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.workflows.prehandle;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.node.app.ServicesAccessor;
+import com.hedera.node.app.service.admin.FreezePreTransactionHandler;
+import com.hedera.node.app.service.consensus.ConsensusPreTransactionHandler;
+import com.hedera.node.app.service.contract.ContractPreTransactionHandler;
+import com.hedera.node.app.service.file.FilePreTransactionHandler;
+import com.hedera.node.app.service.network.NetworkPreTransactionHandler;
+import com.hedera.node.app.service.scheduled.SchedulePreTransactionHandler;
+import com.hedera.node.app.service.token.CryptoPreTransactionHandler;
+import com.hedera.node.app.service.token.TokenPreTransactionHandler;
+import com.hedera.node.app.service.util.UtilPreTransactionHandler;
+import com.hedera.node.app.spi.PreHandleContext;
+import com.hedera.node.app.spi.meta.TransactionMetadata;
+import com.hedera.node.app.state.HederaState;
+import com.hederahashgraph.api.proto.java.TransactionBody;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A {@code PreHandleDispatcher} takes a validated transaction and dispatches it to the correct
+ * handler
+ */
+public final class PreHandleDispatcher {
```

Comment:
Shouldn't this be an interface in `hedera-app-api` and there should be a class called `PreHandleDispatcherImpl` in `hedera-app` ?

Reply:
Right now, it is just an implementation detail of `PreHandleWorkflowImpl` and used nowhere else. Do you think it is useful elsewhere?